index.js

```diff
@@ -111,21 +119,28 @@ app.get('/movies/directors/:director', passport.authenticate('jwt', { session: f
 });
 
 // Get all users
-app.get('/users', passport.authenticate('jwt', { session: false }), (req, res) => {
-    Users.find()
-        .then((users) => {
-            res.status(201).json(users);
-        })
-        .catch((err) => {
-            console.error(err);
-            res.status(500).send('Error: ' + err);
-        });
-});
+// There should never be a route like this in your app
+// app.get('/users', passport.authenticate('jwt', { session: false }), (req, res) => {
+//     Users.find()
+//         .then((users) => {
+//             res.status(201).json(users);
+//         })
+//         .catch((err) => {
+//             console.error(err);
+//             res.status(500).send('Error: ' + err);
+//         });
+// });
 
+/**
+ * Passing the username in the url isnt the best.
+ * The backend will reject if token is not there or inactive
+ * so just use the decoded token info stored in req.user
+ */
 // Get specific user
-app.get('/users/:Username', passport.authenticate('jwt', { session: false }), (req, res) => {
-    Users.findOne({ Username: req.params.Username })
+app.get('/user', passport.authenticate('jwt', { session: false }), (req, res) => {
```

Comment:
When you have this middleware `passport.authenticate('jwt', { session: false })` it takes the token and decodes it and saves it to `req.user` now at this point you should only allow the user fetch their own information unless the aim to so any user to view other users information and even at that you shouldn't send the password back hash or plain